src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -212,22 +312,46 @@ class SyscallHandler {
   int64_t writev(int64_t fd, const void* iovdata, int iovcnt);
 
  private:
-  /** Pointer reference to SimOS object. */
-  SimOS* os_ = nullptr;
-
-  /** Returns the correct Dirfd depending on the pathname and dirfd given to
+  /** Returns the correct dirFd depending on the pathname and dirFd given to
    * syscall. */
   uint64_t getDirFd(int64_t dfd, std::string pathname);
 
   /** If the given filepath points to a special file, the filepath is replaced
    * to point to the SimEng equivalent. */
   std::string getSpecialFile(const std::string filename);
 
+  /** The user-space processes running above the kernel. */
+  const std::unordered_map<uint64_t, std::shared_ptr<Process>>& processes_;
+
+  /** A shared pointer to the simulation memory. */
+  std::shared_ptr<simeng::memory::Mem> memory_;
+
+  /** A callback function to send a syscall result back to a core. */
+  std::function<void(simeng::OS::SyscallResult)> returnSyscall_;
+
+  /** A callback function to get the system time. */
+  std::function<uint64_t()> getSystemTime_;
+
+  /** A callback function to translate a virtual address. */
+  std::function<uint64_t(uint64_t, uint64_t)> vAddrTranslation_;
+
+  /** A callback function to a HostBackedFileMMaps instance's mapfd function. */
+  std::function<HostFileMMap(int, size_t, off_t)> mmapHostFd_;
```

Comment:
I think the function type should be a typedef, maybe in the VMA.hh such that it can be commented on and documented. it will also help communicate intent clearly. It could be `MMapFileOnHost` or whatever you feel is better

Reply:
typedef'ed the function